js/coindcx.js

```diff
@@ -0,0 +1,596 @@
+            'bidVolume': undefined,
+            'ask': this.safeFloat (ticker, 'ask'),
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': this.safeFloat (ticker, 'change_24_hour'),
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeFloat (ticker, 'volume'),
+            'quoteVolume': undefined,
+        };
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = 500, params = {}) {
+        // https://coindcx-official.github.io/rest-api/?shell#candles
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOHLCV requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        if (!(symbol in this.markets)) {
+            throw new ExchangeError (this.id + ' has no symbol ' + symbol);
+        }
+        const market = this.market (symbol);
+        const marketInfo = this.safeValue (market, 'info');
+        const coindcxPair = this.safeString (marketInfo, 'pair');
+        if (coindcxPair === undefined) {
+            throw new ExchangeError (this.id + ' has no pair (look at market\'s info) value for ' + symbol);
+        }
+        const coindcxTimeframe = this.timeframes[timeframe];
+        if (coindcxTimeframe === undefined) {
+            throw new ExchangeError (this.id + ' has no "' + timeframe + '" timeframe');
+        }
+        const request = {
+            'pair': coindcxPair,
+            'interval': coindcxTimeframe,
+            'limit': limit,
+        };
+        const response = await this.publicGetMarketDataCandles (this.extend (request, params));
+        return this.parseOHLCVs (response, market, timeframe, since, limit);
+    }
+
+    parseOHLCV (ohlcv, market = undefined, timeframe = '1m', since = undefined, limit = undefined) {
+        return [
+            this.safeInteger (ohlcv, 'time'),
+            this.safeFloat (ohlcv, 'open'),
+            this.safeFloat (ohlcv, 'high'),
+            this.safeFloat (ohlcv, 'low'),
+            this.safeFloat (ohlcv, 'close'),
+            this.safeFloat (ohlcv, 'volume'),
+        ];
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = 30, params = {}) {
+        // https://coindcx-official.github.io/rest-api/?shell#trades
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchTrades requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        if (!(symbol in this.markets)) {
+            throw new ExchangeError (this.id + ' has no symbol ' + symbol);
+        }
+        const market = this.market (symbol);
+        const marketInfo = this.safeValue (market, 'info');
+        const coindcxPair = this.safeString (marketInfo, 'pair');
```

Comment:
Instead of using the market['info'] here we should use the `market['id']`, as explained in the contributing doc, in the manual and as it is implemented in the other exchanges:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sending-market-ids
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#parsing-symbols
- https://github.com/ccxt/ccxt/wiki/Manual#symbols-and-market-ids

Reply:
About info. On this exchange we have two valuable "ids": a 'symbo'l and a 'pair'. So, I prefer to use the 'symbol' as id, as the exchange uses it more often. In a few cases exchange uses the 'pair' instead of the 'symbol', so I call it from the info. It's ok in this case?

Reply:
Yep, in that case it's ok.